samza-test/src/test/java/org/apache/samza/test/processor/TestZkLocalApplicationRunner.java

```diff
@@ -1011,7 +1011,7 @@ public void testAgreeingOnSameRunIdForBatch() throws InterruptedException {
     LocalApplicationRunner localApplicationRunner1 = (LocalApplicationRunner) appRunner1;
     LocalApplicationRunner localApplicationRunner2 = (LocalApplicationRunner) appRunner2;
 
-    assertEquals("RunId of the two processors does not match", localApplicationRunner2.getRunid(), localApplicationRunner1.getRunid());
+    assertEquals("RunId of the two processors does not match", localApplicationRunner2.getRunId().orElse(null), localApplicationRunner1.getRunId().orElse(null));
```

Comment:
You don't have to unwrap the optional for assertEquals. 
I think it should work out of the box with object comparison.